tests/src/python/test_qgseditformconfig.py

```diff
@@ -57,6 +62,33 @@ def testReadWriteXml(self):
         self.assertFalse(config2.labelOnTop(0))
         self.assertTrue(config2.labelOnTop(1))
 
+    def testFormUi(self):
+        layer = self.createLayer()
+        config = layer.editFormConfig()
+
+        uiLocal = os.path.join(TEST_DATA_DIR, 'layer_attribute_form.ui')
+        (ok, _) = config.setUiForm(uiLocal)
+        self.assertTrue(ok)
+        self.assertEqual(config.layout(), QgsEditFormConfig.UiFileLayout)
+        self.assertEqual(config.uiForm(QgsEditFormConfig.Original), uiLocal)
+        self.assertEqual(config.uiForm(QgsEditFormConfig.LocalCopy), uiLocal)
+
+        uiUrl = 'https://raw.githubusercontent.com/3nids/QGIS/7ac09b1f9b1921e7f95ab077d97dcf0dfd57be4a/tests/testdata/layer_attribute_form.ui'
```

Comment:
Any chance to have this use a minimal local server.

Something like https://stackoverflow.com/a/39801780/2319028

Reply:
is it worth the effort?

Also, `setFormUi` is testing if it exists as a local file first. So would that even work?

Reply:
I think we should minimize the dependency on a particular platform (or we'll never move to gitlab ;) )

Reply:
+1 to @m-kuhn's comment